tests/unit/multi-currency/compatibility/test-class-woocommerce-product-add-ons.php

```diff
@@ -536,6 +557,9 @@ public function test_get_item_data_returns_percentage_price_data_correctly() {
 		];
 
 		$this->mock_multi_currency->method( 'get_price' )->with( 10, 'product' )->willReturn( 10.00 );
-		$this->assertSame( $expected, $this->woocommerce_product_add_ons->get_item_data( [], $addon, $cart_item ) );
+		$actual = $this->woocommerce_product_add_ons->get_item_data( [], $addon, $cart_item );
+		$this->assertSame( $expected['name'], $actual['name'] );
+		$this->assertSame( $expected['value'], strip_tags( $actual['value'] ) );
+		$this->assertSame( $expected['display'], $actual['display'] );
 	}
```

Comment:
I think we can introduce a util to strip_tags for all values of an array, and then we compare it with the `$expected` rather than inspect array item of `$expected`.


```suggestion
		$this->assertSame( $expected, $this->array_strip_tags( $actual ) );
	}

	/**
	 * Strip HTML tags from all values in an array while preserving keys.
	 *
	 * @param array $array The array to process.
	 * @return array The array with HTML tags stripped from all values.
	 */
	private function array_strip_tags( array $array ): array {
		$result = [];
		foreach ( $array as $key => $value ) {
			if ( is_array( $value ) ) {
				$result[ $key ] = $this->array_strip_tags( $value );
			} elseif ( is_string( $value ) ) {
				$result[ $key ] = strip_tags( $value );
			} else {
				$result[ $key ] = $value;
			}
		}
		return $result;
	}
```

Reply:
Yes makes sense, followed your suggestion here [44be7fe](https://github.com/Automattic/woocommerce-payments/pull/11034/commits/44be7fe3ce4284bfc86ab6cced30960ab84cb828)

A heads-up, a fix was just introduced in https://github.com/Automattic/woocommerce-payments/pull/11033 but I prefer the approach of not asserting the mark-up as it feels more robust to future changes.